File: updated_code/code/python/1399-count-largest-group.py
Python
def count_largest_group(n: int) -> int:
    """
    This function counts the number of groups that have the largest size.
    
    Parameters:
    n (int): The input integer.
    
    Returns:
    int: The number of groups that have the largest size.
    """
    freq = {}
    max_size = 0
    max_count = 0
    
    for i in range(1, n + 1):
        dig_sum = sum(int(digit) for digit in str(i))
        
        if dig_sum not in freq:
            freq[dig_sum] = 1
        else:
            freq[dig_sum] += 1
        
        if freq[dig_sum] > max_count or (freq[dig_sum] == max_count and dig_sum > max_size):
            max_size = dig_sum
            max_count = freq[dig_sum]
    
    return max_count

File: updated_code/code/java/1399-count-largest-group.java
/**
 * Package: Programming
 * Description: Count the largest group that sums to equal the digits of their numbers
 *
 * @author [Your Name]
 * @version 1.0
 */
package Programming;

import java.util.Arrays;

public class Solution {
    /**
     * Count the largest group that sums to equal the digits of their numbers
     *
     * @param n the input integer
     * @return the number of groups that have the largest size
     */
    public int countLargestGroup(int n) {
        int[] count = new int[37];
        int max = 0;
        for (int i = 1; i <= n; i++) {
            int sum = 0;
            for (char c : Integer.toString(i).toCharArray()) {
                sum += c - '0';
            }
            count[sum]++;
            max = Math.max(max, count[sum]);
        }
        return Arrays.stream(count).filter(x -> x == max).count();
    }
}